1.11.0: build fails

Build fails with `-Werror=maybe-uninitialized` errors.
gcc 11.1.1
```console
make[2]: Entering directory '/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/x86_64-redhat-linux-gnu'
[ 27%] Building CXX object googlemock/CMakeFiles/shared_gmock_test_.dir/test/gmock-spec-builders_test.cc.o
cd /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/x86_64-redhat-linux-gnu/googlemock && /usr/bin/g++ -DGTEST_LINKED_AS_SHARED_LIBRARY=1 -I/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include -I/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock -I/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include -I/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest -O2 -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fdata-sections -ffunction-sections -DNDEBUG -Wall -Wshadow -Werror -Wno-error=dangling-else -DGTEST_HAS_PTHREAD=1 -fexceptions -MD -MT googlemock/CMakeFiles/shared_gmock_test_.dir/test/gmock-spec-builders_test.cc.o -MF CMakeFiles/shared_gmock_test_.dir/test/gmock-spec-builders_test.cc.o.d -o CMakeFiles/shared_gmock_test_.dir/test/gmock-spec-builders_test.cc.o -c /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc
In file included from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/internal/gtest-death-test-internal.h:39,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-death-test.h:41,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest.h:64,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/internal/gmock-internal-utils.h:47,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-actions.h:145,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h:73,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:35:
In member function ‘testing::internal::MatcherBase<T>& testing::internal::MatcherBase<T>::operator=(const testing::internal::MatcherBase<T>&) [with T = const std::tuple<int>&]’,
    inlined from ‘testing::Matcher<const std::tuple<int>&>& testing::Matcher<const std::tuple<int>&>::operator=(const testing::Matcher<const std::tuple<int>&>&)’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-matchers.h:479:7,
    inlined from ‘testing::internal::OnCallSpec<F>& testing::internal::OnCallSpec<F>::With(const testing::Matcher<const typename testing::internal::Function<F>::ArgumentTuple&>&) [with F = void(int)]’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h:326:20,
    inlined from ‘virtual void {anonymous}::OnCallSyntaxTest_WithIsOptional_Test::TestBody()’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:260:12:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-matchers.h:310:5: error: ‘<unnamed>.testing::Matcher<const std::tuple<int>&>::<unnamed>.testing::internal::MatcherBase<const std::tuple<int>&>::buffer_’ may be used uninitialized [-Werror=maybe-uninitialized]
  310 |     buffer_ = other.buffer_;
      |     ^~~~~~~
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc: In member function ‘virtual void {anonymous}::OnCallSyntaxTest_WithIsOptional_Test::TestBody()’:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:260:12: note: ‘<anonymous>’ declared here
  260 |       .With(_)
In file included from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock.h:65,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:42:
In constructor ‘testing::internal::NaggyMockImpl<MockClass>::NaggyMockImpl() [with Base = {anonymous}::MockA]’,
    inlined from ‘testing::NaggyMock<MockClass>::NaggyMock() [with MockClass = {anonymous}::MockA]’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-nice-strict.h:186:27,
    inlined from ‘void {anonymous}::GMockVerboseFlagTest::TestUninterestingCallOnNaggyMock(bool)’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:2176:22:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-nice-strict.h:120:60: error: ‘a’ may be used uninitialized [-Werror=maybe-uninitialized]
  120 |   NaggyMockImpl() { ::testing::Mock::WarnUninterestingCalls(this); }
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:35:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h: In member function ‘void {anonymous}::GMockVerboseFlagTest::TestUninterestingCallOnNaggyMock(bool)’:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h:438:15: note: by argument 1 of type ‘const void*’ to ‘static void testing::Mock::WarnUninterestingCalls(const void*)’ declared here
  438 |   static void WarnUninterestingCalls(const void* mock_obj)
      |               ^~~~~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:2176:22: note: ‘a’ declared here
 2176 |     NaggyMock<MockA> a;
      |                      ^
In file included from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock.h:65,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:42:
In constructor ‘testing::internal::NaggyMockImpl<MockClass>::NaggyMockImpl() [with Base = {anonymous}::MockB]’,
    inlined from ‘testing::NaggyMock<MockClass>::NaggyMock() [with MockClass = {anonymous}::MockB]’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-nice-strict.h:186:27,
    inlined from ‘virtual void {anonymous}::FunctionCallMessageTest_UninterestingCallOnNaggyMockPrintsArgumentsAndReturnValue_Test::TestBody()’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:2092:20:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-nice-strict.h:120:60: error: ‘b’ may be used uninitialized [-Werror=maybe-uninitialized]
  120 |   NaggyMockImpl() { ::testing::Mock::WarnUninterestingCalls(this); }
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:35:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h: In member function ‘virtual void {anonymous}::FunctionCallMessageTest_UninterestingCallOnNaggyMockPrintsArgumentsAndReturnValue_Test::TestBody()’:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h:438:15: note: by argument 1 of type ‘const void*’ to ‘static void testing::Mock::WarnUninterestingCalls(const void*)’ declared here
  438 |   static void WarnUninterestingCalls(const void* mock_obj)
      |               ^~~~~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:2092:20: note: ‘b’ declared here
 2092 |   NaggyMock<MockB> b;
      |                    ^
In file included from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/internal/gtest-death-test-internal.h:39,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-death-test.h:41,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest.h:64,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/internal/gmock-internal-utils.h:47,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-actions.h:145,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h:73,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:35:
In copy constructor ‘testing::internal::MatcherBase<T>::MatcherBase(const testing::internal::MatcherBase<T>&) [with T = int]’,
    inlined from ‘testing::Matcher<int>::Matcher(const testing::Matcher<int>&)’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-matchers.h:479:7,
    inlined from ‘testing::internal::MockSpec<{anonymous}::Result(int)> {anonymous}::MockA::gmock_ReturnResult(const testing::Matcher<int>&)’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:152:3,
    inlined from ‘virtual void {anonymous}::OnCallSyntaxTest_WithCanAppearAtMostOnce_Test::TestBody()’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:267:3:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-matchers.h:302:33: error: ‘<unnamed>.testing::Matcher<int>::<unnamed>.testing::internal::MatcherBase<int>::buffer_’ may be used uninitialized [-Werror=maybe-uninitialized]
  302 |       : vtable_(other.vtable_), buffer_(other.buffer_) {
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:45:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc: In member function ‘virtual void {anonymous}::OnCallSyntaxTest_WithCanAppearAtMostOnce_Test::TestBody()’:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:268:28: note: ‘<anonymous>’ declared here
  268 |     ON_CALL(a, ReturnResult(_))
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-spi.h:220:48: note: in definition of macro ‘EXPECT_NONFATAL_FAILURE’
  220 |       if (::testing::internal::AlwaysTrue()) { statement; }\
      |                                                ^~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h:2033:3: note: in expansion of macro ‘GMOCK_ON_CALL_IMPL_’
 2033 |   GMOCK_ON_CALL_IMPL_(obj, InternalDefaultActionSetAt, call)
      |   ^~~~~~~~~~~~~~~~~~~
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:268:5: note: in expansion of macro ‘ON_CALL’
  268 |     ON_CALL(a, ReturnResult(_))
      |     ^~~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/internal/gtest-death-test-internal.h:39,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-death-test.h:41,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest.h:64,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/internal/gmock-internal-utils.h:47,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-actions.h:145,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h:73,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:35:
In member function ‘testing::internal::MatcherBase<T>& testing::internal::MatcherBase<T>::operator=(const testing::internal::MatcherBase<T>&) [with T = const std::tuple<int>&]’,
    inlined from ‘testing::Matcher<const std::tuple<int>&>& testing::Matcher<const std::tuple<int>&>::operator=(const testing::Matcher<const std::tuple<int>&>&)’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-matchers.h:479:7,
    inlined from ‘testing::internal::OnCallSpec<F>& testing::internal::OnCallSpec<F>::With(const testing::Matcher<const typename testing::internal::Function<F>::ArgumentTuple&>&) [with F = {anonymous}::Result(int)]’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h:326:20,
    inlined from ‘virtual void {anonymous}::OnCallSyntaxTest_WithCanAppearAtMostOnce_Test::TestBody()’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:267:3:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-matchers.h:310:5: error: ‘<unnamed>.testing::Matcher<const std::tuple<int>&>::<unnamed>.testing::internal::MatcherBase<const std::tuple<int>&>::buffer_’ may be used uninitialized [-Werror=maybe-uninitialized]
  310 |     buffer_ = other.buffer_;
      |     ^~~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:45:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc: In member function ‘virtual void {anonymous}::OnCallSyntaxTest_WithCanAppearAtMostOnce_Test::TestBody()’:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:269:14: note: ‘<anonymous>’ declared here
  269 |         .With(_)
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-spi.h:220:48: note: in definition of macro ‘EXPECT_NONFATAL_FAILURE’
  220 |       if (::testing::internal::AlwaysTrue()) { statement; }\
      |                                                ^~~~~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/internal/gtest-death-test-internal.h:39,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-death-test.h:41,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest.h:64,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/internal/gmock-internal-utils.h:47,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-actions.h:145,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h:73,
                 from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:35:
In member function ‘testing::internal::MatcherBase<T>& testing::internal::MatcherBase<T>::operator=(const testing::internal::MatcherBase<T>&) [with T = const std::tuple<int>&]’,
    inlined from ‘testing::Matcher<const std::tuple<int>&>& testing::Matcher<const std::tuple<int>&>::operator=(const testing::Matcher<const std::tuple<int>&>&)’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-matchers.h:479:7,
    inlined from ‘testing::internal::OnCallSpec<F>& testing::internal::OnCallSpec<F>::With(const testing::Matcher<const typename testing::internal::Function<F>::ArgumentTuple&>&) [with F = {anonymous}::Result(int)]’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/include/gmock/gmock-spec-builders.h:326:20,
    inlined from ‘virtual void {anonymous}::OnCallSyntaxTest_WithCanAppearAtMostOnce_Test::TestBody()’ at /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:267:3:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-matchers.h:310:5: error: ‘<unnamed>.testing::Matcher<const std::tuple<int>&>::<unnamed>.testing::internal::MatcherBase<const std::tuple<int>&>::buffer_’ may be used uninitialized [-Werror=maybe-uninitialized]
  310 |     buffer_ = other.buffer_;
      |     ^~~~~~~
In file included from /home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:45:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc: In member function ‘virtual void {anonymous}::OnCallSyntaxTest_WithCanAppearAtMostOnce_Test::TestBody()’:
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googlemock/test/gmock-spec-builders_test.cc:270:14: note: ‘<anonymous>’ declared here
  270 |         .With(_)
/home/tkloczko/rpmbuild/BUILD/googletest-release-1.11.0/googletest/include/gtest/gtest-spi.h:220:48: note: in definition of macro ‘EXPECT_NONFATAL_FAILURE’
  220 |       if (::testing::internal::AlwaysTrue()) { statement; }\
      |                                                ^~~~~~~~~
cc1plus: all warnings being treated as errors
```

--- a/googletest/include/gtest/gtest-matchers.h
+++ b/googletest/include/gtest/gtest-matchers.h
@@ -299,17 +299,18 @@ class MatcherBase : private MatcherDescriberInterface {
   }
 
  protected:
-  MatcherBase() : vtable_(nullptr) {}
+  MatcherBase() : vtable_(nullptr), buffer_() {}
 
   // Constructs a matcher from its implementation.
   template <typename U>
-  explicit MatcherBase(const MatcherInterface<U>* impl) {
+  explicit MatcherBase(const MatcherInterface<U>* impl)
+      : vtable_(nullptr), buffer_() {
     Init(impl);
   }
 
   template <typename M, typename = typename std::remove_reference<
                             M>::type::is_gtest_matcher>
-  MatcherBase(M&& m) {  // NOLINT
+  MatcherBase(M&& m) : vtable_(nullptr), buffer_() {  // NOLINT
     Init(std::forward<M>(m));
   }
 

